Assets/MixedRealityToolkit/Interfaces/Registrars/IMixedRealityServiceRegistrar.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Interfaces
+{
+    /// <summary>
+    /// Interface for Mixed Reality Toolkit service registration.
+    /// </summary>
+    public interface IMixedRealityServiceRegistrar
+    {
+        #region IMixedRealityService registration
+
+        /// <summary>
+        /// Registers a service of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be registered (ex: IMixedRealityBoundarySystem).</typeparam>
+        /// <param name="serviceInstance">An instance of the service to be registered.</param>
+        bool RegisterService<T>(T serviceInstance) where T : IMixedRealityService;
+
+        /// <summary>
+        /// Registers a service of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be registered (ex: IMixedRealityBoundarySystem).</typeparam>
+        /// <param name="concreteType">The concrete type to instantiate.</param>
+        /// <param name="supportedPlatforms">The platform(s) on which the service is supported.</param>
+        /// <param name="args">Optional arguments used when instantiating the concrete type.</param>
+        /// <returns>True if the service was successfully registered, false otherwise.</returns>
+        bool RegisterService<T>(
+            Type concreteType,
+            SupportedPlatforms supportedPlatforms = (SupportedPlatforms)(-1),
+            params object[] args) where T : IMixedRealityService;
+
+        /// <summary>
+        /// Unregisters a service of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be unregistered (ex: IMixedRealityBoundarySystem).
+        /// <param name="name">The name of the service to unregister.</param>
+        /// <returns>True if the service was successfully unregistered, false otherwise.</returns>
+        /// <remarks>If the name argument is not especified, the first instance will be unregistered</remarks>
```

Comment:
Out of curiousity, what's the use case for not specifying the name? I read this comment as saying that things will be unregistered in FIFO pattern. I can see the coordination here being tricky (i.e. this allows for scenarios where the order in which different components call unregister can get tricky).

Reply:
core systems are un-named.

one of the goals of this pr is to limit behavioral change in the service locator. 

the team is definitely willing to entertain thoughts of how to make improvements to the existing locator, feel free to file issues and they can be addressed in the not too distant future